swap_meet/vendor.py

```diff
@@ -1,2 +1,64 @@
+from .item import Item
+
+
 class Vendor:
-    pass
\ No newline at end of file
+    def __init__(self, inventory=None):
+        if inventory:
+            self.inventory = inventory
+        else:
+            self.inventory = []
+
+    def add(self, item):
+        self.inventory.append(item)
+        return item
+
+    def remove(self, item):
+        if item in self.inventory:
+            self.inventory.remove(item)
+            return item
+
+        return False
+
+    def get_by_category(self, category):
+        result = [item for item in self.inventory if item.category is category]
+        return result
+
+    def get_best_by_category(self, category):
+        all_by_category = self.get_by_category(category)
+
+        if not all_by_category:
+            return None
+
+        max_rating = max(item.condition for item in all_by_category)
+        best_by_category = [
+            item for item in all_by_category if item.condition == max_rating
+        ]
+        # Assumed that in event of tie, return first item with highest rating
+        return best_by_category[0]
+
+    def swap_items(self, other_vendor, my_item, their_item):
+        if not all([my_item in self.inventory, their_item in other_vendor.inventory]):
```

Comment:
Great exploration of what we can do with the `all()` function! This becomes really handy as the number of conditional statements we're checking grows. In this case, I think it might be easier to read as individual conditionals in the if-statement using the `not in` operator:
```suggestion
        if my_item not in self.inventory or their_item not in other_vendor.inventory:
```